Add Project: mkdocs-landing

TEParsons opened the pull request:
**What kind of change does this PR introduce?**
- [x] Add a project
- [ ] Update a project
- [ ] Remove a project
- [ ] Add or update a category
- [ ] Change configuration
- [ ] Documentation
- [ ] Other, please describe:

**Description:**
Adds the mkdocs theme "Landing" to the catalogue - a theme designed for making personal landing pages. See [Landing's website](https://teparsons.github.io/mkdocs-landing) for documentation and example implementations.

**Checklist:**
- [x] I have read the [CONTRIBUTING](https://github.com/best-of-lists/best-of/blob/main/CONTRIBUTING.md) guidelines.
- [x] I have not modified the `README.md` file. Projects are only supposed to be added or updated within the `projects.yaml` file since the `README.md` file is automatically generated.

pawamoy reviewed (approved):
Thanks!

TEParsons commented:
Thanks for pulling this in :) So this will become visible when `update-best-of-list` is next run, right?

pawamoy commented:
Yes, it happens once a week. I can retrigger it manually today though :+1:

TEParsons commented:
Hmm, looks like the preview isn't working, do you need to rebuild mkdocs-gallery too?

pawamoy commented:
Rebuilding the gallery right now.

pawamoy commented:
Erm, playwright doesn't work anymore on my system...

pawamoy commented:
I'm getting this error:

```pytb
Traceback (most recent call last):
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/mkdocs/__main__.py", line 288, in build_command
    build.build(cfg, dirty=not clean)
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/mkdocs/commands/build.py", line 337, in build
    _build_page(
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/mkdocs/commands/build.py", line 223, in _build_page
    output = template.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/jinja2/environment.py", line 1304, in render
    self.environment.handle_exception()
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/jinja2/environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/mkdocs_landing/main.html", line 12, in top-level template code
    {% include "partials/header_element.html" %}
^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/mkdocs_landing/partials/header_element.html", line 24, in top-level template code
    {{ config.theme.tagline | markdown(id="tagline") }}
^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/mkdocs_landing/filters.py", line 33, in parse_markdown
    html = md.convert(text)
           ^^^^^^^^^^^^^^^^
  File "/media/data/dev/mkdocs-gallery/themes/landing/.venv/lib/python3.12/site-packages/markdown/core.py", line 341, in convert
    if not source.strip():
           ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'strip'
```

Do I have to add extra config to mkdocs.yml?

TEParsons commented:
Ahh, yes, I didn't supply a default for that so I need to handle when it's undefined. But yes if you're able to, just supplying a value for theme.tagline will do the trick. Otherwise I'll fix that in the next version & let you know when it's on pypi.

pawamoy commented:
Thanks!

> Otherwise I'll fix that in the next version & let you know when it's on pypi.

That'd be great, the gallery builds each theme with a common template, to fix that I'd have to special case mkdocs-landing :sweat_smile:

TEParsons commented:
Okay, I've released a new version! Here's the release: https://github.com/TEParsons/mkdocs-landing/releases/tag/0.0.4

I also added a test suite and added a test that a site with no configuration at all builds, which should catch this in future. Could you link me to the specific mkdocs.yaml used to build site previews in the gallery? I'll add it to my test suite (and implement the same sort of thing for Torillic too :P )

pawamoy commented:
Great :tada: Let me try to rebuild the gallery.

Sure, here's the template: https://github.com/pawamoy/mkdocs-gallery/blob/main/templates/specimen/mkdocs.yml.

pawamoy commented:
OK, good, preview now working :smile:
